src/modules/template-management/template-form/useTemplateManagement.ts

```diff
@@ -0,0 +1,44 @@
+import { useState, useCallback, useEffect } from 'react';
+import { useHistory } from 'react-router';
+
+import { addTemplate, AddTemplatePayload } from 'core/api';
+
+interface State {
+  pending: boolean;
+  error: string;
+  id: string | null;
+}
+
+type Return = [State, (payload: AddTemplatePayload) => Promise<void>];
+
```

Comment:
Namespace aswell?

Reply:
for useTemplateManagement i need to create namespace - useTemplateManagement which is invalid if you look from ts good practices perspective. Every model should have name starting from big letter - so for hooks we can't use namespace